docs/advanced_usage.rst

```diff
@@ -371,3 +371,36 @@ The resulting prompt changes and looks like::
 
 .. note::
    As you can see the order of the options changed from ``1 - MIT`` to ``1 - Apache Software License 2.0``. **Cookiecutter** takes the first value in the list as the default.
+
+.. _`template extensions`:
+
+Template Extensions
+-------------------
+
+*New in Cookiecutter 1.4*
+
+A template may extend the Cookiecutter environment with custom `Jinja2 extensions`_,
+that can add extra filters, tests, globals or even extend the parser.
+
+To do so, a template author must specify the required extensions in ``cookiecutter.json`` as follows:
+
+.. code-block:: json
+
+    {
+        "project_slug": "Foobar",
+        "year": "{% now 'utc', '%Y' %}",
+        "_extensions": ["jinja2_time.TimeExtension"]
+    }
+
+On invocation Cookiecutter tries to import the extensions and add them to its environment respectively.
+
+In the above example, Cookiecutter provides the additional tag `now`_, after
+installing the `jinja2_time.TimeExtension`_ and enabling it in ``cookiecutter.json``.
+
+Please note that Cookiecutter will **not** install any dependencies on its own!
```

Comment:
I agree, @pydanny.

Tbh, I thought of this PR to be the starting point for a number of changes that deal with support for extra dependencies. As a next step, I'd like get #429 implemented and I already have a local wip branch to add another hook that will be run before environment initialization.

If the user agrees to run this hook, it could create a new virtualenv and install the dependencies of the template or abort generation altogether.